### ISSUE TITLE
Warn against here-links in doc comments

### ISSUE THREAD

**Issue by SabrinaJewson** [rust-lang/rust-clippy#13480]
### What it does

Warns when a user writes `[here]`, `[click here]`, `[link]` or `[this link]` in a doc comment. This lint would be in the “Style” category, as I think it is fairly universally agreed-upon that such links should be avoided (I have not seen arguments to the contrary, at least).

### Advantage

Links of this form are widely considered bad for accessibility[[1]](https://www.w3.org/QA/Tips/noClickHere)[[2]](https://ux.stackexchange.com/questions/12100/why-shouldnt-we-use-words-such-as-here-and-this-in-textlinks)[[3]](https://granicus.com/blog/why-click-here-links-are-bad/), among other reasons. It is recommended to change the link text to a short description of what is at the URL instead.

### Drawbacks

Sometimes it is non-obvious how to rewrite a here-link into the preferred form, which could increase friction when writing docs.

### Example

```rust
/// For more details on this function, click [here](crate).
```

Could be written as:

```rust
/// For more details on this function, see [the crate-level docs](crate).
```
